tests/ui/impl-trait/precise-capturing/capture-parent-arg.rs

```diff
@@ -0,0 +1,32 @@
+#![feature(precise_capturing)]
+//~^ WARN the feature `precise_capturing` is incomplete
+
+trait Tr {
+    type Assoc;
+}
+
+struct W<'a>(&'a ());
+
+impl Tr for W<'_> {
+    type Assoc = ();
+}
+
+impl<'a> W<'a> {
+    fn good1() -> impl use<'a> Into<<W<'a> as Tr>::Assoc> {}
+}
+
+impl<'a> W<'a> {
+    fn good2() -> impl use<'a> Into<<Self as Tr>::Assoc> {}
```

Comment:
Document that this test is ensuring that `Self` captures work, even though they refer to the parent's `'a` instead of the opaque type's copy